go-controller/pkg/cluster/master.go

```diff
@@ -255,6 +256,9 @@ func (cluster *OvnClusterController) addNode(node *kapi.Node) (err error) {
 		}
 		logrus.Infof("Allocated node %s HostSubnet %s", node.Name, hostsubnet.String())
```

Comment:
Shouldn't there be a `break` here? If we successfully allocated we don't want to keep running through the allocator list...

(the actual fixes all look good afaict)

Reply:
@danwinship good catch on the `break` thing.  It is an orthogonal issue, will submit a separate PR for this.